### PR TITLE
feat: initial implementation of sync apb

### PR DESCRIPTION
**Pull request by darahayes**
This is the initial implementation of the data-sync-apb. It was created by taking the metrics-apb and refactoring it.

### What does this include?

- [x] Provision Task
	* data-sync-ui (behind an openshift oauth proxy)
	* data-sync-server
	* postgres
- [x] Deprovision Task
- [x] Verify database connectivity
- [x] Verify that the bind action results with config showing up in mobile services.json
- [x] Verify the unbind results in config being removed from mobile-services.json
- [x] Initial Implementation of deprovision and provision tests

## Todo

- [x] Verify the tests
- [x] Push image to aerogearcatalog
- [ ] Documentation
- [x] Set up Jenkins

### Provisioned Services
screen shot 2018-07-06 at 13 04 12

### Client Config
screen shot 2018-07-06 at 12 35 34

### CODE REVIEW

**Comment by aliok** [comment]
APB provision worked nicely. Will continue my review on Monday

**Comment by pb82** [comment]
Provision, Bind, Unbind and Unprovision all work. Retrieving the config via mobile-cli works too:

```
⇒  mobile get clientconfig myapp-android --namespace=myproject -o json
{
	"version": 1,
	"clusterName": "https://192.168.178.20:8443",
	"namespace": "myproject",
	"clientId": "myapp-android",
	"services": [
		{
			"id": "data-sync-myapp-android",
			"name": "data-sync",
			"type": "data-sync",
			"url": "https://data-sync-server-myproject.192.168.178.20.nip.io",
			"config": {}
		}
	]
}
```

File: docs/modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+* xref:index.adoc[Overview]
```
Comment (pb82) [inline docs/modules/ROOT/nav.adoc]: We can remove everything from the docs structure (it's only meant for technical non-user facing docs)

File: docs/modules/ROOT/pages/app_metrics_guide.adoc [hunk 2]
```diff
@@ -0,0 +1,83 @@
+= Analyzing App usage using the metrics service
```
Comment (pb82) [inline docs/modules/ROOT/pages/app_metrics_guide.adoc]: All the docs pages can be deleted. We can keep the directories and boilerplate for non-user facing docs.

File: roles/bind-data-sync-apb/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+- name: Retrieve metrics route
+  shell: "oc get configmaps/{{ sync_secret_name }} -n {{ namespace }} -o jsonpath={.data.uri}"
+  register: sync_server_uri_result
+
+- set_fact:
+    sync_server_uri: "{{ sync_server_uri_result.stdout }}"
+
+- name: "Create metrics bind secret yaml file"
```
Comment (pb82) [inline roles/bind-data-sync-apb/tasks/main.yml]: metrics -> sync

File: roles/provision-data-sync-apb/tasks/provision-sync-server.yml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  register: data_sync_server_route
+
+# Create the configmap used for the client config
+- name: Create metrics configmap template
+  template:
+    src: client-configmap.yml.j2
+    dest: /tmp/client-configmap.yaml
+  when: _apb_service_instance_id is defined
+
+- name: Create metrics test configmap template
```
Comment (pb82) [inline roles/provision-data-sync-apb/tasks/provision-sync-server.yml]: metrics -> sync

File: roles/provision-data-sync-apb/tasks/provision-sync-server.yml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      app: data-sync
+      service: data-sync-server
+    to_name: data-sync-server
+    spec_port_target_port: web
+    tls_termination: edge
+    tls_insecure_edge_termination_policy: Redirect
+  register: data_sync_server_route
+
+# Create the configmap used for the client config
+- name: Create metrics configmap template
```
Comment (pb82) [inline roles/provision-data-sync-apb/tasks/provision-sync-server.yml]: metrics -> sync

File: roles/provision-data-sync-apb/tasks/provision-sync-server.yml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  template:
+    src: client-configmap.yml.j2
+    dest: /tmp/client-configmap.yaml
+  when: _apb_service_instance_id is defined
+
+- name: Create metrics test configmap template
+  template:
+    src: client-configmap-test.yml.j2
+    dest: /tmp/configmap.yaml
```
Comment (pb82) [inline roles/provision-data-sync-apb/tasks/provision-sync-server.yml]: This needs to be `client-configmap.yaml` in order for the tests to pass (The next tasks expects that filename)

**Comment by aliok** [comment]
It is a bit hard to review the Ansible code, but looks good to me.
Let's merge after fixing the CI build

**Review by aliok — APPROVED** [review]

**Comment by darahayes** [comment]
@aliok I'm just about to push one more commit with some minor fixes to the tests. The CI isn't actually failing because of the test though, there's a problem with the CI instance.

**Comment by pb82** [comment]
@darahayes please also remove all the remaining metrics documentation. The actual docs will land in the mobile-docs repo.

**Comment by pb82** [comment]
@darahayes nvm. i see it's already done.

**Pull request merged**